docs/authentication.md

```diff
@@ -33,22 +33,62 @@ sequenceDiagram
     opt user doesn't have access
         AuthService ->> User: Message with access denied
     end
-    loop Refresh backend connection every 3 minutes
-        AuthService ->> Backend: Refresh token (GET auth/refresh)
-        Backend ->> AuthService: Response with new backend access token in cookies
+    loop Check backend availability every 3 minutes
+        AuthService ->> Backend: (GET /)
+        Backend ->> AuthService: Response with 'Hello World!'
+    end
+    loop Refresh backend token every backend access token lifetime
+        AuthService ->> Backend: Refresh token (GET auth/keycloak)
+        Backend ->> AuthService: Response with new backend access token in cookies and new accessExp
     end
-    loop Refresh token every 1/3 of tokens lifetime
+    loop Refresh keycloak token every 1/3 of keycloak access tokens lifetime
         AuthService ->> Keycloak: Refresh token
-        Keycloak ->> AuthService: Updated token
+        Keycloak ->> AuthService: Updated access token and refresh token
     end
     User ->> AuthService: Logout
+    AuthService ->> Keycloak: Logout (GET /auth/realms/{our-realm}/protocol/openid-connect/logout)
     AuthService ->> Backend: Invalidate session (DELETE /auth/logout)
     Backend ->> AuthService: Response with cookies deleted
-    AuthService ->> Keycloak: Logout
     AuthService ->> User: Clear user data
 ```
 
-## Non-Keycloak
+-   **User requests login**: User initiates login with SSO.
```

Comment:
You could use numbered list (best with manually hardcoded numbers), then those numbers could correspond to numbers of items on the diagram